src/util/mapbox.js

```diff
@@ -113,5 +129,13 @@ function parseUrl(url: string): UrlObject {
 
 function formatUrl(obj: UrlObject): string {
     const params = obj.params.length ? `?${obj.params.join('&')}` : '';
-    return `${obj.protocol}://${obj.authority}${obj.path}${params}`;
+    const url = `${obj.protocol}://${obj.authority}${obj.path}${params}`;
+    return transformURL(url);
 }
+
+exports.resolveURL = function(url: string): string {
+    url = transformURL(url);
+    const a = window.document.createElement('a');
+    a.href = url;
+    return transformURL(a.href);
```

Comment:
We're now transforming the URL twice, once before resolving, and once after resolving. `resolveURL` is only used for GeoJSON sources that can reference GeoJSON files relative to the web page's location. Using the same `transformURL` call means that they sometimes get relative URLs (pre-resolving), and sometimes resolved URLs with a full scheme and hostname. It also means that we're calling `transformURL` twice for a single URL when it's a GeoJSON source, compared to only once for all other URLs.